sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,157 @@
+    'sklearn.setup',
+    'sklearn.svm',
+    'sklearn.utils',
+    # Deprecated modules
+    'sklearn.cross_validation',
+    'sklearn.grid_search',
+    'sklearn.learning_curve',
+])
+
+# functions to ignore args / docstring of
+_DOCSTRING_IGNORES = [
+    'sklearn.utils.deprecation.load_mlcomp',
+    'sklearn.pipeline.make_pipeline',
+    'sklearn.pipeline.make_union',
+    'sklearn.utils.extmath.safe_sparse_dot',
+]
+
+# Methods where y param should be ignored if y=None by default
+_METHODS_IGNORE_NONE_Y = [
+        'fit',
+        'score',
+        'fit_predict',
+        'fit_transform',
+        'partial_fit',
+        'predict'
+]
+
+
+def test_docstring_parameters():
+    # Test module docstring formatting
+
+    # Skip test if numpydoc is not found or if python version is < 3.5
+    try:
+        import numpydoc  # noqa
+        assert sys.version_info >= (3, 5)
+    except (ImportError, AssertionError):
+        raise SkipTest(
+            "numpydoc is required to test the docstrings")
+
+    from numpydoc import docscrape
+
+    incorrect = []
+    for name in PUBLIC_MODULES:
+        with warnings.catch_warnings(record=True):
+            module = importlib.import_module(name)
+        classes = inspect.getmembers(module, inspect.isclass)
+        for cname, cls in classes:
+            this_incorrect = []
+            if cname in _DOCSTRING_IGNORES:
+                continue
+            if cname.startswith('_') and cname not in _DOC_SPECIAL_MEMBERS:
+                continue
+            with warnings.catch_warnings(record=True) as w:
+                cdoc = docscrape.ClassDoc(cls)
+            if len(w):
+                raise RuntimeError('Error for __init__ of %s in %s:\n%s'
+                                   % (cls, name, w[0]))
+
+            cls_init = getattr(cls, '__init__', None)
+
+            if _is_deprecated(cls_init):
+                continue
+
+            elif cls_init is not None:
+                this_incorrect += check_parameters_match(cls.__init__, cdoc,
+                                                         class_name=cname)
+            for method_name in cdoc.methods:
+                method = getattr(cls, method_name)
+                if _is_deprecated(method):
+                    continue
+                param_ignore = None
+                # Now skip docstring test for y when y is None
+                # by default for API reason
+                if method_name in _METHODS_IGNORE_NONE_Y:
+                    sig = signature(method)
+                    if ('y' in sig.parameters and
+                            sig.parameters['y'].default is None):
+                        param_ignore = ['y']  # ignore y for fit and score
+                result = check_parameters_match(method, ignore=param_ignore,
+                                                class_name=cname)
+                this_incorrect += result
+
+            if hasattr(cls, '__call__') and not _is_deprecated(cls.__call__):
```

Comment:
Can't find your comment on github, but you only skip things that start with an underscore and are not ing ``_DOC_SPECIAL_MEMBERS`` but ``__call__`` is. So again, why is this not covered above?